test/neo4j_test.py

```diff
@@ -74,18 +74,89 @@ def test_construct_pdp_query():
     # The fractions should all add up to around 100 percent
     assert sum_percent == pytest.approx(100)
 
+def test_construct_pdp_query_return_values():
+    """
+    Test that the construct_pdp_query function returns the expected query for a
+    known graph. These tests will not actually execute the query
+    """
+    q1 = textwrap.dedent("""\
+            MATCH path = (n0:Compound)-[:BINDS_CbG]-(n1)-[:PARTICIPATES_GpPW]-(n2)-[:PARTICIPATES_GpPW]-(n3)-[:ASSOCIATES_DaG]-(n4:Disease)
+            USING JOIN ON n2
+            WHERE n0.identifier = { source }
+            AND n4.identifier = { target }
+            AND n1 <> n3
+            WITH
+            [
+            size((n0)-[:BINDS_CbG]-()),
+            size(()-[:BINDS_CbG]-(n1)),
+            size((n1)-[:PARTICIPATES_GpPW]-()),
+            size(()-[:PARTICIPATES_GpPW]-(n2)),
+            size((n2)-[:PARTICIPATES_GpPW]-()),
+            size(()-[:PARTICIPATES_GpPW]-(n3)),
+            size((n3)-[:ASSOCIATES_DaG]-()),
+            size(()-[:ASSOCIATES_DaG]-(n4))
+            ] AS degrees, path
+            WITH path, reduce(pdp = 1.0, d in degrees| pdp * d ^ -{ w }) AS PDP
+            WITH collect({paths: path, pdps: PDP}) AS allData, sum(PDP) AS DWPC
+            UNWIND allData AS data
+            WITH data.paths AS path, data.pdps AS PDP, DWPC
+            RETURN
+              substring(reduce(s = '', node IN nodes(path)| s + '–' + node.name), 1) AS path,
+              PDP,
+              100 * (PDP / DWPC) AS PERCENT_OF_DWPC
+            ORDER BY PERCENT_OF_DWPC DESC
+            """).rstrip()
+
+    dwpc = 0.03287590886921623
+    q2 = textwrap.dedent("""\
+            MATCH path = (n0:Compound)-[:BINDS_CbG]-(n1)-[:PARTICIPATES_GpPW]-(n2)-[:PARTICIPATES_GpPW]-(n3)-[:ASSOCIATES_DaG]-(n4:Disease)
+            USING JOIN ON n2
+            WHERE n0.identifier = { source }
+            AND n4.identifier = { target }
+            AND n1 <> n3
+            WITH
+            [
+            size((n0)-[:BINDS_CbG]-()),
+            size(()-[:BINDS_CbG]-(n1)),
+            size((n1)-[:PARTICIPATES_GpPW]-()),
+            size(()-[:PARTICIPATES_GpPW]-(n2)),
+            size((n2)-[:PARTICIPATES_GpPW]-()),
+            size(()-[:PARTICIPATES_GpPW]-(n3)),
+            size((n3)-[:ASSOCIATES_DaG]-()),
+            size(()-[:ASSOCIATES_DaG]-(n4))
+            ] AS degrees, path
+            WITH path, reduce(pdp = 1.0, d in degrees| pdp * d ^ -{ w }) AS PDP
+            RETURN
+            substring(reduce(s = '', node IN nodes(path)| s + '–' + node.name), 1) AS path,
+            PDP,
+            100 * (PDP / 0.03287590886921623) AS PERCENT_OF_DWPC
+            ORDER BY PERCENT_OF_DWPC DESC
+            """).rstrip()
+
+    # Set up the graph for querying
+    directory = pathlib.Path(__file__).parent.absolute()
+    path = directory.joinpath('data/bupropion-CbGpPWpGaD-subgraph.json.gz')
```

Comment:
Need to update to the `hetionet-v1.0-metagraph.json` path. This works but downloads more than is necessary.